packages/block-editor/src/components/image-size-control/index.js

```diff
@@ -33,6 +33,37 @@ export default function ImageSizeControl( {
 	const { currentHeight, currentWidth, updateDimension, updateDimensions } =
 		useDimensionHandler( height, width, imageHeight, imageWidth, onChange );
 
+	/**
+	 * Updates the dimensions for the given scale.
+	 * Handler for toggle group control change.
+	 *
+	 * @param {number} scale The scale to update the dimensions for.
+	 */
+	const handleUpdateDimensions = ( scale ) => {
+		if ( undefined === scale ) {
+			updateDimensions();
+			return;
+		}
+
+		const scaledWidth = Math.round( imageWidth * ( scale / 100 ) );
+		const scaledHeight = Math.round( imageHeight * ( scale / 100 ) );
+
+		updateDimensions( scaledHeight, scaledWidth );
+	};
+
+	/**
+	 * Handler for adding the value to toggle group control.
+	 */
+	const addSelectedScaleValue = () => {
+		const scaleSize = Math.round( currentWidth * ( 100 / imageWidth ) );
+
+		if ( IMAGE_SIZE_PRESETS.includes( scaleSize ) ) {
+			return scaleSize;
+		}
+
+		return undefined;
+	};
```

Comment:
I think we need to follow [the original logic](https://github.com/WordPress/gutenberg/blob/98f3f6a5b3cccfcc8d94cefa9a4bee9750478132/packages/block-editor/src/components/image-size-control/index.js#L83-L85) and check that not only the width but also the height matches.

We should be able to determine this as follows:

```javascript
const selectedValue = IMAGE_SIZE_PRESETS.find( ( scale ) => {
	const scaledWidth = Math.round( imageWidth * ( scale / 100 ) );
	const scaledHeight = Math.round( imageHeight * ( scale / 100 ) );
	return currentWidth === scaledWidth && currentHeight === scaledHeight;
} );
```

Reply:
Thanks @t-hamano, For the updated logic, I would add this logic and would wait for `ToggleGroupControl` bug to be fixed 🤞.

Reply:
Update: The logic has been updated and waiting for the ToggleGroupControl fix to be land.

Thank You,